Userland/Libraries/LibJS/Bytecode/Pass/LoadElimination.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2022, Leon Albrecht <leon.a@serenityos.com>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Bitmap.h>
+#include <AK/TypeCasts.h>
+#include <LibJS/Bytecode/Op.h>
+#include <LibJS/Bytecode/PassManager.h>
+
+namespace JS::Bytecode::Passes {
+
+static NonnullOwnPtr<BasicBlock> eliminate_loads(BasicBlock const& block, size_t number_of_registers)
+{
+    auto array_ranges = Bitmap::must_create(number_of_registers, false);
+
+    for (auto it = InstructionStreamIterator(block.instruction_stream()); !it.at_end(); ++it) {
+        if ((*it).type() == Instruction::Type::NewArray) {
+            Op::NewArray const& array_instruction = static_cast<Op::NewArray const&>(*it);
+            if (size_t element_count = array_instruction.element_count())
+                array_ranges.set_range<true, false>(array_instruction.start().index(), element_count);
+        }
+    }
+
+    auto new_block = BasicBlock::create(block.name(), block.size());
+    HashMap<size_t, Register> identifier_table {};
+    HashMap<u32, Register> register_table {};
+
+    for (auto it = InstructionStreamIterator(block.instruction_stream()); !it.at_end();) {
```

Comment:
Please at least add some comments explaining what this pass is detecting and what it's doing to them. as thorough and detailed as possible :^)

Reply:
Is the updated amount of comments enough?